nncf/torch/extensions/__init__.py

```diff
@@ -68,8 +76,30 @@ def get(self, fn_name: str) -> Callable:
         :return: A callable object corresponding to the requested function.
         """
         if self._loaded_namespace is None:
+            time_limit = int(os.environ.get(NNCF_TIME_LIMIT_TO_LOAD_EXTENSION, 60))
+
+            def extension_loader_timeout_handler(signum, frame):
+                # pylint: disable=line-too-long
+                msg = textwrap.dedent(
+                    f"""\
+                    The extension load function failed to execute within {time_limit} seconds.
+                    To resolve this issue, run the following command:
+                        rm -rf {self._loader.get_build_dir()}
+                    For a machine with poor performance, you may try increasing the time limit by setting the environment variable:
+                        {NNCF_TIME_LIMIT_TO_LOAD_EXTENSION}=180
+                    More information about reasons read on https://github.com/openvinotoolkit/nncf/blob/develop/docs/FAQ.md#importing-anything-from-nncftorch-hangs
+                    """
+                )
+                raise ExtensionLoaderTimeoutException(msg)
+
             with extension_is_loading_info_log(self._loader.name()):
-                self._loaded_namespace = self._loader.load()
+                signal.signal(signal.SIGALRM, extension_loader_timeout_handler)
```

Comment:
This is not cross-platform. We should find another way or abandon this effort whatsoever.

Reply:
Changed to use `ThreadPool` and `apply_async` to get extension from thread. , check on windows - works.